packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -1,46 +1,63 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Button } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
-import { withDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
 
-const SettingsHeader = ( {
-	openDocumentSettings,
-	openBlockSettings,
-	sidebarName,
-} ) => {
+const SettingsHeader = ( { sidebarName } ) => {
+	// translators: Label for the Block Settings Sidebar tab, not selected.
 	const blockLabel = __( 'Block' );
+
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+
+	const documentLabel = useSelect( ( select ) => {
+		const currentPostType = select( 'core/editor' ).getCurrentPostType();
+		const postType = select( 'core' ).getPostType( currentPostType );
+
+		// translators: Default ARIA label for the Document sidebar tab, not selected.
+		const defaultDocumentLabel = __( 'Document' );
+
+		return get(
+			postType,
+			[ 'labels', 'singular_name' ],
+			defaultDocumentLabel
+		);
+	} );
+
 	const [ documentAriaLabel, documentActiveClass ] =
 		sidebarName === 'edit-post/document'
 			? // translators: ARIA label for the Document sidebar tab, selected.
-			  [ __( 'Document (selected)' ), 'is-active' ]
-			: // translators: ARIA label for the Document sidebar tab, not selected.
-			  [ __( 'Document' ), '' ];
+			  [ sprintf( __( '%1$s (selected)' ), documentLabel ), 'is-active' ]
+			: [ documentLabel, '' ];
 
 	const [ blockAriaLabel, blockActiveClass ] =
 		sidebarName === 'edit-post/block'
-			? // translators: ARIA label for the Settings Sidebar tab, selected.
-			  [ __( 'Block (selected)' ), 'is-active' ]
-			: // translators: ARIA label for the Settings Sidebar tab, not selected.
-			  [ __( 'Block' ), '' ];
+			? // translators: ARIA label for the Block Settings Sidebar tab, selected.
+			  [ sprintf( __( '%1$s (selected)' ), blockLabel ), 'is-active' ]
```

Comment:
Let's avoid the interpolation, I think it contributes to complexity, loss of context, dependency between i18n strings. So let's keep `__( 'Block (selected) ' )`.

If you prefer to avoid the inlined value, you can always define `const blockSelectedLabel` next to `blockLabel`. If so, then I might even suggest turning them into `BLOCK_LABEL` and `BLOCK_SELECTED_LABEL`, to make it clearer that they are placeholders for the actual string.